tests/blackbox/testlib/utils.py

```diff
@@ -53,16 +56,20 @@ def random_string(length=4):
 
 def process_exists(name):
     """
-    Walk the process table looking for executable 'name', returns pid if one
-    found, else return None
+    Look through processes, using their pids, to find one matching 'name'.
+    Return None if no such process found, else return the pid.
+    :param name: name of process to check
+    :type name: string
+    :return: pid or None
+    :rtype: int or NoneType
     """
-    for pid in [pid for pid in os.listdir("/proc") if pid.isdigit()]:
+    for proc in psutil.process_iter():
```

Comment:
If you do use ```process_iter``` you might as well get the benefit. I think there is some design where you can specify that it should just get the name, and not all the information about the process, which is far more efficient. You might as well try that out.